File: Algo/hr_implement/5/beautifulTriplets.cpp
/*Erica wrote an increasing sequence of n numbers in her notebook. She considers a triplet to be beautiful if:
i < j < k
arr[j] - arr[i] = arr[k] - arr[j] == d */

#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int beautifulTriplets(int d, vector<int> arr)
{
    int count = 0;
    for (int i = arr.size() - 1; i > 0; i--)
    {
        for (int j = i - 1; j >= 0; j--)
        {
            if (arr[i] - arr[j] == d)
            {
                for (int k = j - 1; k >= 0; k--)
                {
                    if (arr[j] - arr[k] == d)
                    {
                        count++;
                        cout << "yo=> " << arr[i] << "-" << arr[j] << endl;
                    }
                }
            }
        }
    }
    return count;
}

int main()
{
    vector<int> arr;
    arr.push_back(1);
    arr.push_back(2);
    arr.push_back(4);
    arr.push_back(5);
    arr.push_back(7);
    arr.push_back(8);
    arr.push_back(10);
    beautifulTriplets(3, arr);
    return 0;
}

File: Algo/hr_implement/5/beautifulTriplets.js
function beautifulTriplets(d, a) {
  let count = 0;
  for (let i = a.length - 1; i > 0; i--) {
    for (let j = i - 1; j > -1; j--) {
      if (a[i] - a[j] == d) {
        for (let k = j - 1; k > -1; k--) {
          if (a[j] - a[k] == d) {
            count++;
          }
        }
      }
    }
  }
  return count;
}

let a = [1, 2, 4, 5, 7, 8, 10];
console.log(beautifulTriplets(3, a));
